lib/util/git/modified-cache.ts

```diff
@@ -22,18 +21,16 @@ export function setCachedModifiedResult(
   isModified: boolean
 ): void {
   const cache = getCache();
-  cache.branches ??= [];
-  const { branches } = cache;
-  const branch =
-    branches?.find((branch) => branch.branchName === branchName) ??
-    ({ branchName: branchName } as BranchCache);
+  const branch = cache.branches?.find(
+    (branch) => branch.branchName === branchName
+  );
 
-  // if branch not present add it to cache
-  if (is.undefined(branch.sha)) {
-    branches.push(branch);
+  if (!branch) {
+    logger.debug(`Branch cache not present for ${branchName}`);
```

Comment:
Will this log every time when repoCache is not enabled?

Reply:
No. A minimal branchState is created always which will prevent this.